tests/wp-tests/BM/Metabox/Posts/CategoryTest.php

```diff
@@ -220,5 +220,49 @@ public function test_remove_category_from_posts_without_overwrite() {
 		// Assert that category 2 has one posts.
 		$posts_in_cat2 = $this->get_posts_by_category( $cat2 );
 		$this->assertEquals( count( $posts_in_cat2 ), 1 );
-	}	
+	}
+	
+	/**
+	 * Test remove category from post with overwrite.
+	 */
+	public function test_remove_category_from_posts_with_overwrite() {
+		// Create two categories and a common category.
+		$cat1 = $this->factory->category->create( array( 'name' => 'cat1' ) );
+		$cat2 = $this->factory->category->create( array( 'name' => 'cat2' ) );
+		$common_cat = $this->factory->category->create( array( 'name' => 'common_cat' ) );
+
+		// Create one post in each category.
+		// The first post will also have the common category.
+		$post1 = $this->factory->post->create( array( 'post_title' => 'post1', 'post_category' => array( $cat1, $common_cat ) ) );
+		$post2 = $this->factory->post->create( array( 'post_title' => 'post2', 'post_category' => array( $cat2 ) ) );
+
+		// Assert that each category has one post.
+		$posts_in_cat1 = $this->get_posts_by_category( $cat1 );
+		$posts_in_cat2 = $this->get_posts_by_category( $cat2 );
+		$posts_in_common_cat = $this->get_posts_by_category( $common_cat );
+
+		$this->assertEquals( count( $posts_in_cat1 ), 1 );
+		$this->assertEquals( count( $posts_in_cat2 ), 1 );
+		$this->assertEquals( count( $posts_in_common_cat ), 1 );
+
+		// Invoke our method.
+		$options = array(
+			'old_cat'   => $cat1,
+			'new_cat'   => -1,
+			'overwrite' => true,
+		);
+		$this->category_metabox->move( $options );
+
+		// Assert that category 1 has no posts.
```

Comment:
Even for this test case you have not updated the comments.

Having wrong comments is worse than not having comments at all.

Reply:
Test case comments changed.